proto-lens-protoc/src/Data/ProtoLens/Setup.hs

```diff
@@ -209,10 +216,41 @@ generateSources root l files = do
                      [ InstalledPackageInfo.dataDir info </> protoLensImportsPrefix
                      | info <- collectDeps l
                      ]
-    generateProtosWithImports (root : importDirs) (autogenModulesDir l)
+    -- Generate .hs files into a temporary directory, then move them over
+    -- to the target (autogen) directory only if they are different from
+    -- what's already there. This way, we don't needlessly touch the generated
+    -- .hs files when nothing changes, and thus don't needlessly make GHC
+    -- recompile them (as it considers their modification times for that).
+    -- See https://github.com/google/proto-lens/issues/176
+    let tmpAutogenModulesDir = autogenModulesDir l ++ "-protoc-tmpOutDir"
+    -- Generate .hs files into temp dir.
+    generateProtosWithImports (root : importDirs) tmpAutogenModulesDir
                               -- Applying 'root </>' does nothing if the path is already
                               -- absolute.
                               (map (root </>) files)
+    -- Discover generated files.
+    -- `getDirectoryContentsRecursive` is lazy IO, so we `force` through
+    -- the list to make it strict hereinafter.
+    !generatedFiles <- force <$> getDirectoryContentsRecursive tmpAutogenModulesDir
```

Comment:
I am curious if you do this out of general principle that lazy IO is bad or because there's an observable difference.

Reply:
The former (to make for least-surprising code)

Reply:
I'm fairly certain it makes no difference here, here because the very next thing you do is call `forM_` on the result.

Normally it wouldn't matter much, but in this case you're adding a new dependency (`deepseq`) so I don't think it's worth it unless you've benchmarked it (or, perhaps, used the generated Core to proved that this way is better).